server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -892,6 +892,7 @@ public UserVm resetVMSSHKey(ResetVMSSHKeyCmd cmd) throws ResourceUnavailableExce
     }
 
     protected void removeEncryptedPasswordFromUserVmVoDetails(UserVmVO userVmVo) {
+        _vmDao.loadDetails(userVmVo);
```

Comment:
@weizhouapache I'm not sure but instead of loading all details, removing one detail and then storing all details, can we do:
```
        userVmDetailsDao.removeDetail(userVmVo.getId(), VmDetailConstants.ENCRYPTED_PASSWORD);
        _vmDao.loadDetails(userVmVo);
```
Also in the caller of this method we load details (ln 859), not sure if that is needed

Reply:
@shwstppr 
user vm details is updated in line 883 (ssh key is added), but `userVmVo` does not contain it.
`boolean result = resetVMSSHKeyInternal(vmId, sshPublicKey);`
therefore we need to reload the details. 
it is good idea to use `userVmDetailsDao.removeDetail`. I will change the code and test it. thanks.

Reply:
@weizhouapache yes, VM details are updated by `resetVMSSHKeyInternal` and that is why I feel `loadDetails` at 859 can be removed as I don't see it being used anywhere.

Reply:
@shwstppr removed line 859.